autoload/lsp/lsp.vim

```diff
@@ -764,12 +764,14 @@ enddef
 # Display the hover message from the LSP server for the current cursor
 # location
 export def Hover(cmdmods: string)
-  var lspserver: dict<any> = buf.CurbufGetServerChecked('hover')
-  if lspserver->empty()
+  var lspservers: list<dict<any>> = buf.CurbufGetServerCheckedList('hover')
+  if lspservers->empty()
     return
   endif
 
-  lspserver.hover(cmdmods)
+  for lspserver in lspservers
+    lspserver.hover(cmdmods)
```

Comment:
If one LSP server provides the hover information, then there is no need to query other servers for hover information.

Reply:
Doesn't thst check for hover for the whole document and not the current location? Since different LSPs can give more or less relevant info depending on location, each needs to be queried?

Reply:
How will we display the hover information in a popup from different LSP servers?  We can display the information from only one LSP server.

Reply:
That is something I still haven't solved. In my example above, angular-ls gave a reply when I triggered hover on the html-code that is inside a string. And the reason there is also a warning about documentation not found, that is typescript-ls answering. The same with the other example where typescript-ls answered with a good reply but angular-ls answered documentation not found.

In those cases, the easiest solution is probably to just nog show the warning. if no LSP gives an answer, perhaps not show anything at all? Perhaps that is fine?

When two or more LSPs actually gives an answer, now you probably have to go by priority and pick the first LSP added to your config. And that is how it works today I believe?

I think there are a few good reasons why someone might want to use several LSPs on the same document. I personally would be helped by having typescript and angular. I know I probably would want to have typescript and eslint diagnostics on the same document. I still haven't gotten eslint to work, but when it works I would have syntax diagnostics from typescript and code style diagnostics from eslint. This would be very useful. But for that to work you have to get the response from all the LSPs, not just the first one.